### DISCUSSION TITLE
Permissions constraints generates error messages

### DISCUSSION THREAD

**Discussion by qutach** [netbox-community/netbox#9034] [Q&A] [answered]
### NetBox version

v3.1.10

### Python version

3.9

### Steps to Reproduce

We are trying to configure constrains for the object type "IPAM > IP address".
Enter the following in Constraints box:
`{"vrf__in": ["blue-1", "red-2"]}`

This generates an error message
`Field 'id' expected a number but got 'blue-1'.`

Works if we use `{"vrf_id__in": [1, 2]}`

### Expected Behavior

That it works with using vrf names in constrains for object IP Adresses

### Observed Behavior

![bild1](https://user-images.githubusercontent.com/47136329/161749796-541d5906-df83-4576-bb25-cdfb94166efb.png)
![bild2](https://user-images.githubusercontent.com/47136329/161749812-69a7414b-dc9f-4dbe-a95a-e06e15b6e316.png)

**Comment by jeremystretch** [comment, marked as answer]
> `Field 'id' expected a number but got 'blue-1'.`

Just as the error message implies, what you've written is not a valid filter. Filtering on the `vrf` foreign key requires a numeric ID (or list of IDs when using `in`). If you want to filter by the VRF name, you need to specify that:

`{"vrf__name__in": ["blue-1", "red-2"]}`

You can find much more information on how querysets work in the [Django documentation](https://docs.djangoproject.com/en/4.0/ref/models/querysets/).
